Code/Source/Frame/NamespaceConfiguration.h

```diff
@@ -0,0 +1,49 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#pragma once
+
+#include <AzCore/std/string/string.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    /// Configuration for handling of namespaces
+    struct NamespaceConfiguration
+    {
+    public:
+        AZ_RTTI(NamespaceConfiguration, "{5E5BC6EA-DD01-480E-A4D1-6857CF70FDC8}");
+        NamespaceConfiguration() = default;
+        virtual ~NamespaceConfiguration() = default;
+
+        enum NamespaceStrategy
+        {
+            Default, // FromEntityName for top-level frames, Empty otherwise
+            Empty,
+            FromEntityName,
+            Custom   // Non-empty but unrelated to entity name
+        };
+
+        void PopulateNamespace(bool isTop, const AZStd::string &entityName);
+        AZStd::string GetNamespace(const AZStd::string& parentNamespace) const;
+
+        static void Reflect(AZ::ReflectContext* context);
+
+    private:
+        AZStd::string m_namespace;
+        NamespaceStrategy m_namespaceStrategy;
+        bool m_isTop;
```

Comment:
Little confused by this variable name. Maybe `m_isRoot` would fit it better?

Reply:
isTopLevel? Or isRoot?